src/test/java/se/sundsvall/relations/integration/db/RelationRepositoryTest.java

```diff
@@ -60,61 +63,92 @@ void create() {
 				.withNamespace(targetNamespace).build())
 			.build();
 
+		final var inverseEntity = RelationEntity.builder()
+			.withType(type.getCounterType())
+			.withMunicipalityId(municipalityId)
+			.withSource(relationEntity.getTarget())
+			.withTarget(relationEntity.getSource())
+			.withInverseRelation(relationEntity)
+			.build();
+
+		relationEntity.setInverseRelation(inverseEntity);
+
 		final var persistedEntity = repository.save(relationEntity);
 
 		assertThat(persistedEntity).isNotNull();
 		assertThat(persistedEntity.getId()).isNotNull();
-		assertThat(persistedEntity.getType()).isEqualTo(type);
+		assertThat(persistedEntity.getType().getName()).isEqualTo("type-1");
 		assertThat(persistedEntity.getMunicipalityId()).isEqualTo(municipalityId);
 		assertThat(persistedEntity.getCreated()).isCloseTo(OffsetDateTime.now(), within(2, SECONDS));
 		assertThat(persistedEntity.getModified()).isNull();
+		assertThat(persistedEntity.getSource().getId()).isNotNull();
 		assertThat(persistedEntity.getSource().getResourceId()).isEqualTo(sourceId);
 		assertThat(persistedEntity.getSource().getType()).isEqualTo(sourceType);
 		assertThat(persistedEntity.getSource().getService()).isEqualTo(sourceService);
 		assertThat(persistedEntity.getSource().getNamespace()).isEqualTo(sourceNamespace);
+		assertThat(persistedEntity.getTarget().getId()).isNotNull();
 		assertThat(persistedEntity.getTarget().getResourceId()).isEqualTo(targetId);
 		assertThat(persistedEntity.getTarget().getType()).isEqualTo(targetType);
 		assertThat(persistedEntity.getTarget().getService()).isEqualTo(targetService);
 		assertThat(persistedEntity.getTarget().getNamespace()).isEqualTo(targetNamespace);
-
+		// Inverse
+		assertThat(persistedEntity.getInverseRelation().getId()).isNotNull();
+		assertThat(persistedEntity.getInverseRelation().getCreated()).isCloseTo(OffsetDateTime.now(), within(2, SECONDS));
+		assertThat(persistedEntity.getInverseRelation()).isSameAs(inverseEntity);
+		assertThat(persistedEntity.getInverseRelation().getInverseRelation()).isSameAs(persistedEntity);
+		assertThat(persistedEntity.getSource().getId()).isEqualTo(persistedEntity.getInverseRelation().getTarget().getId());
+		assertThat(persistedEntity.getTarget().getId()).isEqualTo(persistedEntity.getInverseRelation().getSource().getId());
 	}
 
 	@Test
 	void update() {
 
-		final var entity = repository.findById("1");
+		final var entity = repository.findById("1").orElseThrow(() -> new RuntimeException("Error in test data"));
 		final var newSourceNamespace = "newSourceNamespace";
-		final var newTyp = "newType";
+		final var newTyp = typeRepository.findById("rt3").orElseThrow(() -> new RuntimeException("Error in test data"));
```

Comment:
newType?

Reply:
It will become the new type for the relation once updated.